File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
   
    var counts = tangerine.reduce((pv,cv)=>{
            pv[cv] = (pv[cv] || 0) + 1;
            return pv;
        }, {});
        
    const arrCount = Object.values(counts).sort((a,b) => b-a);
    
    let sum=0;
    let answer=0;
    
    for(let cnt of arrCount) {
        sum+=cnt;
        answer++;
        if(sum>=k) break;
    }
        
    
   
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    dic = dict()
    
    for i in range(len(tangerine)) :
        dic[tangerine[i]] = dic.get(tangerine[i], 0) + 1
        
    sorted_dic = dict(sorted(dic.items(), key=lambda x: x[1], reverse = True))
    
    
    for (key, item) in sorted_dic.items() :
        k-= item
        answer+=1
        
        if k <=0 :
            return answer
    
    return answer